src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/linalg/IterativeLinearSolverNDArray.java

```diff
@@ -0,0 +1,183 @@
+package org.broadinstitute.hellbender.tools.coveragemodel.linalg;
+
+import org.apache.commons.math3.linear.NonPositiveDefiniteOperatorException;
+import org.broadinstitute.hellbender.tools.coveragemodel.SubroutineSignal;
+import org.broadinstitute.hellbender.utils.param.ParamUtils;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.factory.Nd4j;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+/**
+ * A basic implementation for iterative linear system solvers
```

Comment:
Again, a brief mathematical exposition (or even just least a link to the relevant wiki page) could be useful.  Basically, just any documentation that can help a developer decide when this class might come in handy.